programs/survival/easier_renewable_sponge.sc

```diff
@@ -0,0 +1,29 @@
+__on_player_attacks_entity(player,entity)->(
+	schedule(0,'__add_dead_fish',entity)//if not the fish's health isn't 0 and I cant check if its dead or not
+);
+global_puff=0;
+global_trop=0;
+global_salmon=0;
+global_cod=0;
+
+__add_dead_fish(entity)->(
+	if (entity=='Pufferfish'&& query(entity,'health')==0,
+		global_puff+=1;
+	);
+	if (entity=='Tropical Fish'&& query(entity,'health')==0,
+		global_trop+=1;
+	);
+	if (entity=='Salmon'&& query(entity,'health')==0,
+		global_salmon+=1;
+	);
+	if (entity=='Cod'&& query(entity,'health')==0,
+		global_cod+=1;
+	);
+	if(global_puff>0 && global_trop >0 && global_salmon>0 && global_cod>0 &&rand(50)==0,
```

Comment:
I assume you will start getting spongs when you kill at least one of each?
if you would keep player stats in a map, then  you could use all(...) function. But that would reset with every restart - I assume that's intended

Reply:
ok yeah, I have no clue clue how to use any form of map. I always end up with northe pointing wst and south pointing east

Reply:
in this case map or list, doesn't matter that much, but either would look much better than several if's next to each other.

Reply:
ok, yeah, i have no clue how to do this wth lists either